### PR TITLE
Search with toolbar

### PR DESCRIPTION
**Pull request by whitzhu**
UPDATED SEARCH BUTTON & INPUT: 

screen shot 2016-08-17 at 9 38 38 am
screen shot 2016-08-17 at 9 42 35 am
screen shot 2016-08-17 at 9 43 57 am
screen shot 2016-08-17 at 9 44 16 am
screen shot 2016-08-17 at 9 44 32 am

screen shot 2016-08-15 at 4 43 15 pm

screen shot 2016-08-15 at 4 42 48 pm

*Short description*

## TODO

- [ ] Have tests been written for the new code?
- [ ] Has documentation been written/updated?
- [ ] New dependencies (if any) added to requirements file
- [ ] Add an entry to CHANGELOG.rst
- [ ] Add yourself it AUTHORS.rst if you don't appear there

## Reviewer guidance

*If you PR has a significant size, give the reviewer some helpful remarks*

## Issues addressed

List the issues solved or partly solved by the PR

## Documentation

If the PR has documentation, link the file here (either .rst in your repo or if built on Read The Docs)

## Screenshots (if appropriate)

They're nice. :)

### CODE REVIEW

**Comment by christianmemije** [comment]
@indirectlylit Are there plans of merging this PR anytime soon?

**Comment by indirectlylit** [comment]
I'd started reviewing but there's a lot going on and it needs to be done carefully. Is it blocking further bug fixes?

**Comment by whitzhu** [comment]
@indirectlylit  Updated Search button style 
screen shot 2016-08-17 at 9 38 38 am

screen shot 2016-08-17 at 9 43 57 am
screen shot 2016-08-17 at 9 44 16 am
screen shot 2016-08-17 at 9 44 32 am

**Comment by christianmemije** [comment]
@indirectlylit I was planning on working on this bug https://trello.com/c/OhZJJ1yd/406-bug-possible-to-confuse-learn-toolbar-media-query-state after this got merged.

File: kolibri/plugins/learn/assets/src/vue/search-widget/index.vue [hunk 1]
```diff
@@ -142,60 +148,114 @@
...
+    top: 0.5rem
+    height: 26px
+    width: 60px
+    margin-left: 10px
+    padding: 0.2em 0.7em
+    border-radius: 4px
+    font-size: 0.8em
+    border: 1px solid $core-text-annotation
+    color: $core-text-annotation
+    z-index: 10001
```
Comment (indirectlylit) [inline kolibri/plugins/learn/assets/src/vue/search-widget/index.vue]: noo!

File: kolibri/plugins/learn/assets/src/vue/search-widget/index.vue [hunk 2]
```diff
@@ -4,24 +4,29 @@
...
+            @keyup="search() | debounce 500"
+            @keydown.esc.prevent="clear()">
+        </div>
+        <button aria-label="Reset" class="reset" type="reset" @click="clear()" :style="{ visibility: localSearchTerm ? 'inherit' : 'hidden' }">
         <svg height="24" viewbox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg">
           <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"></path>
           <path d="M0 0h24v24H0z" fill="none"></path>
         </svg>
       </button>
+        <search-button class='search-btn'>Cancel</search-button>
```
Comment (radinamatic) [inline kolibri/plugins/learn/assets/src/vue/search-widget/index.vue]: So many stuff going on here, it's hard to keep up... 

`Cancel` does not sound very clear to me, not sure I've seen it in the search context. What is the goal here? If the purpose is to close the search "overlay", maybe `Close Search` gives a better idea?

File: kolibri/plugins/learn/assets/src/vue/search-widget/index.vue [hunk 3]
```diff
@@ -142,60 +148,114 @@
...
+      display: block
+      width: 100%
+      left: 0
 
   .reset
     border: none
-    background-color: $core-bg-light // IE10 needs a non-transparent bg to be clickable
+    background-color: $core-bg-canvas // IE10 needs a non-transparent bg to be clickable
     display: inline-block
     outline: none
```
Comment (radinamatic) [inline kolibri/plugins/learn/assets/src/vue/search-widget/index.vue:235]: Please add `:focus` style for this button, since outline is set to `none`.

**Pull request merged**